pkg/controller/servingcert/controller/secret_updating_controller_test.go

```diff
@@ -368,6 +368,29 @@ func TestRequiresRegenerationServiceUIDMismatchBetaAnnotation(t *testing.T) {
 			},
 			expected: true,
 		},
+		{
+			name: "secret with added data",
```

Comment:
I don't think this tests your related changes as it is just exercising the check against an already-queued secret to see if it matches up with its owner (yours is good, so returning "no regeneration" is to be expected). You should make a test in the e2e suite that creates a serving cert like normal and then modifies the secret, and then checks to see if it gets stomped on or not.

Reply:
I updated the test, added e2e